server/src/main/java/com/defold/extender/services/CocoaPodsService.java

```diff
@@ -0,0 +1,625 @@
+
+        // vendored_frameworks
+        JSONArray vendored = getAsJSONArray(specJson, "vendored_frameworks");
+        if (vendored != null) {
+            spec.vendoredframeworks.addAll(vendored);
+        }
+
+        // libraries
+        spec.libraries.addAll(getAsJSONArray(specJson, "libraries"));
+        if (ios != null) spec.ios_libraries.addAll(getAsJSONArray(ios, "libraries"));
+        if (osx != null) spec.osx_libraries.addAll(getAsJSONArray(osx, "libraries"));
+
+        // parse subspecs
+        JSONArray subspecs = getAsJSONArray(specJson, "subspecs");
+        if (subspecs != null) {
+            Iterator<JSONObject> it = subspecs.iterator();
+            while (it.hasNext()) {
+                JSONObject o = it.next();
+                PodSpec subSpec = createPodSpec(o, spec, podsDir);
+                spec.subspecs.add(subSpec);
+            }
+        }
+
+        // find source and header files
+        List<String> sourceFilePatterns = new ArrayList<>();
+        JSONArray sourceFiles = getAsJSONArray(specJson, "source_files");
+        if (sourceFiles != null) {
+            Iterator<String> it = sourceFiles.iterator();
+            while (it.hasNext()) {
+                String path = it.next();
+                spec.sourceFiles.addAll(findPodSourceFiles(spec, path));
+                spec.includePaths.addAll(findPodIncludePaths(spec, path));
+                // Cocoapods uses Ruby where glob patterns are treated slightly differently:
+                // Ruby: foo/**/*.h will find .h files in any subdirectory of foo AND in foo/
+                // Java: foo/**/*.h will find .h files in any subdirectory of foo but NOT in foo/
+                if (path.contains("/**/")) {
+                    path = path.replaceFirst("\\/\\*\\*\\/", "/");
+                    spec.sourceFiles.addAll(findPodSourceFiles(spec, path));
+                    spec.includePaths.addAll(findPodIncludePaths(spec, path));
+                }
+            }
+        }
+
+        return spec;
+    }
+
+    private PodSpec createPodSpec(String specJson, File podsDir) throws ExtenderException {
+        PodSpec spec = createPodSpec(parseJson(specJson), null, podsDir);
+        spec.originalJSON = specJson;
+        return spec;
+    }
+
+    /**
+     * Install pods from a podfile and create PodSpec instances for each installed pod.
+     * @param workingDir Directory where to install pods. The directory must contain a valid Podfile
+     * @return A list of PodSpec instances for the installed pods
+     */
+    private List<PodSpec> installPods(File workingDir) throws IOException, ExtenderException {
+        LOGGER.info("Installing pods");
+        File podFile = new File(workingDir, "Podfile");
+        if (!podFile.exists()) {
+            throw new ExtenderException("Unable to find Podfile " + podFile);
+        }
+        File dir = podFile.getParentFile();
+        String log = execCommand("pod install --verbose", workingDir);
```

Comment:
Is it safe to run multiple "pod" commands from different threads?
E.g. will they thrash the cache for each other?

Where is the cache located? If it's "default", do we want it there?
Or do we want it under the extender's temp folder ? E.g. like gradle

Reply:
CocoaPods handle concurrent "pod install" commands: https://github.com/CocoaPods/CocoaPods/pull/10018

Reply:
The default cache location is in `/Users/USER/Library/Caches/CocoaPods`

Reply:
It seems like it is possible to control cache and repo location by setting CP_HOME_DIR:

https://guides.cocoapods.org/using/faq.html#can-i-change-the-default-cocoapods-repositories-folder

Reply:
Yeah, it might be good to specify the cache path ourselves as it might not always be preferable to install it globally (e.g. if we run it ourselves locally)